src/HotChocolate/Language/src/Language.Rewriters/Contracts/ISyntaxNavigator.cs

```diff
@@ -0,0 +1,51 @@
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+
+namespace HotChocolate.Language.Rewriters;
+
+/// <summary>
+/// The syntax navigator keeps track of the syntax path that has been traversed and allows to access the nodes in the path in a streamlined way.
+/// </summary>
+public interface ISyntaxNavigator
+{
+    /// <summary>
+    /// Returns the immediate parent of the current Syntax Node
+    /// </summary>
+    ISyntaxNode? Parent { get; }
+    
+    /// <summary>
+    /// Adds a syntax node to the Syntax Navigator to record the parent of the Syntax Node being visited.
+    /// </summary>
+    /// <param name="node">The parent syntax node to be added to the Syntax Navigator</param>
+    void Push(ISyntaxNode node);
+
+    /// <summary>
+    /// Removes the current parent node from the Syntax Navigator.
+    /// </summary>
+    /// <param name="node">The removed parent node.</param>
+    void Pop(out ISyntaxNode node);
```

Comment:
```suggestion
    ISyntaxNode void Pop();
```

this one now should return the node instead of using out. Its more in line how Microsoft implements a stack